homeassistant/components/vacuum/neato.py

```diff
@@ -140,7 +141,13 @@ def supported_features(self):
     @property
     def battery_level(self):
         """Return the battery level of the vacuum cleaner."""
-        return self._state['details']['charge']
+        if self._available:
+            return self._state['details']['charge']
```

Comment:
I would make an instance attribute `self._battery_level` that we return here and move the update of that to `update`. Then we don't need to do an extra check of availability here.

Reply:
Definitely more efficient :) thank you